### PR TITLE
Fix click on faded left/right/middle panel -> close settings

### PR DESCRIPTION
**Pull request by lukebarnard1**
- implement generic dispatch to close user/room/group settings
 - use dispatch to allow clicks on disabled left/right/middle panel to
   close settings

A much more maintainable approach would be to use dedicate routing
instead of doing different things depending on what page of the app is
currently being viewed. At the very least we could make the concept of a
settings page generic.
da9fe99

### CODE REVIEW

File: src/components/structures/GroupView.js [hunk 1]
```diff
@@ -563,12 +566,22 @@ export default React.createClass({
         this._closeSettings();
     },
 
+    _onAction(payload) {
+        switch (payload.action) {
+            // NOTE: close_settings is an app-wide dispatch; as it is dispatched from MatrixChat
```
Comment (dbkr) [inline src/components/structures/GroupView.js]: Aren't all dispatches app-wide?

Comment (lukebarnard1) [reply src/components/structures/GroupView.js]: Yes. I guess I felt like it was worth mentioning that it's handled in multiple places.

**Comment by lukebarnard1** [comment]
>Yes, I see what you mean, this is a bit grim isn't it.

Yep.

File: src/components/structures/GroupView.js (hunk 1 above)
Comment (dbkr) [reply src/components/structures/GroupView.js]: ah right, I see what you mean.

Comment (lukebarnard1) [reply src/components/structures/GroupView.js]: I'll clarify this.

**Pull request merged**